config/confignet/confignet.go

```diff
@@ -39,11 +44,15 @@ type TCPAddr struct {
 	// If the host is a literal IPv6 address it must be enclosed in square brackets, as in "[2001:db8::1]:80" or
 	// "[fe80::1%zone]:80". The zone specifies the scope of the literal IPv6 address as defined in RFC 4007.
 	Endpoint string `mapstructure:"endpoint"`
+
+	// DialerTimeout is the maximum amount of time a dial will wait for
+	// a connect to complete. The default is no timeout.
+	DialerTimeout time.Duration `mapstructure:"dialer_timeout"`
 }
 
 // Dial equivalent with net.Dial for this address.
 func (na *TCPAddr) Dial() (net.Conn, error) {
```

Comment:
If we are here, should we also change this to accept context?

Reply:
It looks like the `net` authors let `net.Dial` and `net.DialTimeout` use `context.Background()` and expect users to use `net.DialContext()`.  Instead of our wrapper merging the concepts, would it be better to expose a `TCPAddr.DialContext()` that calls `net.DialContext`? That would let us avoid a breaking change.

Reply:
> It looks like the net authors let net.Dial and net.DialTimeout use context.Background() and expect users to use net.DialContext()

They are legacy, because net was added before context. I think we should have a rule of thumb that all our public APIs should accept Context. This is not yet 1.0 so we can fix that here.

Reply:
@bogdandrutu do you want me to make that change in this PR or in a separate PR?

Reply:
Separate PR please.

Reply:
Created https://github.com/open-telemetry/opentelemetry-collector/issues/9105 to track this topic